xdsl/irdl.py

```diff
@@ -955,12 +956,14 @@ def build_arg(arg_def: Any, arg: Any) -> Any:
     return res, arg_sizes
 
 
-def irdl_op_builder(cls: type[_OpT], op_def: OpDef,
-                    operands: Sequence[SSAValue | Operation
-                                       | list[SSAValue | Operation] | None],
-                    res_types: Sequence[Any | list[Any] | None],
-                    attributes: dict[str, Any], successors: Sequence[Block],
-                    regions: Sequence[Any | None]) -> _OpT:
+def irdl_op_builder(
+        cls: type[_OpT], op_def: OpDef,
+        operands: Sequence[SSAValue | Operation
+                           | list[SSAValue | Operation] | list[SSAValue]
```

Comment:
Actually same here, we could use `Sequence` instead of lists.
We should probably modify this code to instead of checking if this is a list, checking if this is not an `SSAValue | Operation`.